TimberAPI/EntityLinkerSystem/EntityLinker.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Timberborn.ConstructibleSystem;
+using Timberborn.Persistence;
+using TimberbornAPI.Internal;
+using UnityEngine;
+
+namespace TimberbornAPI.EntityLinkerSystem
+{
+    /// <summary>
+    /// Defines the behaviour of Linkers
+    /// </summary>
+    public class EntityLinker : MonoBehaviour, IFinishedStateListener, IPersistentEntity
+    {
+        //Keys for saving/loading
+        protected static readonly ComponentKey EntityLinkerKey = new ComponentKey("EntityLinker");
+        protected static readonly ListKey<EntityLink> EntityLinksKey = new ListKey<EntityLink>(nameof(EntityLinks));
+
+        internal readonly List<EntityLink> _entityLinks = new List<EntityLink>();
+        public IReadOnlyCollection<EntityLink> EntityLinks { get; private set; }
+
+        public virtual void Awake()
+        {
+            EntityLinks = _entityLinks.AsReadOnly();
+        }
+
+        /// <summary>
+        /// Save existing Links
+        /// </summary>
+        /// <param name="entitySaver"></param>
+        public virtual void Save(IEntitySaver entitySaver)
+        {
+            IObjectSaver component = entitySaver.GetComponent(EntityLinkerKey);
+            component.Set(EntityLinksKey, EntityLinks);
+        }
+
+        /// <summary>
+        /// Load saves Links
+        /// </summary>
+        /// <param name="entityLoader"></param>
+        public virtual void Load(IEntityLoader entityLoader)
+        {
+            if (!entityLoader.HasComponent(EntityLinkerKey))
+            {
+                return;
+            }
+            IObjectLoader component = entityLoader.GetComponent(EntityLinkerKey);
+            if (component.Has(EntityLinksKey))
+            {
+                var links = component.Get(EntityLinksKey);
+                if (links == null)
+                {
+                    return;
+                }
+                var linkerLinks = links.Where(x => x.Linker == this).ToList();
+                _entityLinks.AddRange(linkerLinks);
+                foreach (var link in linkerLinks)
+                {
+                    PostCreateLink(link);
```

Comment:
Does it need to do this within a load?

Reply:
@KYPremco You're totally correct. I can just _not_ filter the list of links and add them with AddRange() and it will work. Good catch!

Reply:
@KYPremco actually scratch that, seems like it pretty much needs to be this way for the Links to work